File: assets/problems/2023/cola/sol/subtask2.py
cola = "1"*10010

pos = 1
salto = 2
while pos <= 10**4:
    cola = cola[:pos] + "0" + cola[pos+1:]
    pos += salto
    salto += 1

N = int(input())
for _ in range(N):
    X = int(input())
    print(cola[X])
    

File: assets/problems/2023/cola/sol/subtask2.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    string cola = "";
    for (int i = 0; i < 10010; ++i) {
        cola += "1";
    }
    int pos = 1, salto = 2;
    while (pos < 10000) {
        cola[pos] = '0';
        pos += salto;
        salto++;
    }
    int n, x;
    cin >> n;
    while (n--) {
        cin >> x;
        cout << cola[x] << endl;
    }
    return 0;
}